File: go-sdk/pkg/models/operations/deletefolderlocksid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteFolderLocksIDPathParams struct {
	FolderLockID string `pathParam:"style=simple,explode=false,name=folder_lock_id"`
}

type DeleteFolderLocksIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteFolderLocksIDPathParams
}

type DeleteFolderLocksIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/deletefolderlocksid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteFolderLocksIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=folder_lock_id" })
  folderLockId: string;
}


export class DeleteFolderLocksIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteFolderLocksIdPathParams;
}


export class DeleteFolderLocksIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
